include/aspect/velocity_boundary_conditions/ascii_data.h

```diff
@@ -0,0 +1,356 @@
+  Copyright (C) 2011 - 2014 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#ifndef __aspect__velocity_boundary_conditions_ascii_data_h
+#define __aspect__velocity_boundary_conditions_ascii_data_h
+
+#include <aspect/velocity_boundary_conditions/interface.h>
+#include <aspect/simulator_access.h>
+
+
+namespace aspect
+{
+  namespace VelocityBoundaryConditions
+  {
+    using namespace dealii;
+
+
+    namespace internal
+    {
+
+      /**
+       * AsciiDataLookup handles all kinds of tasks around looking up a certain
+       * velocity boundary condition from a text file. This class
+       * keeps around the contents of two sets of files, corresponding to two
+       * instances in time where the text files provides us with data; the boundary
+       * values at one particular time are interpolated between the two
+       * currently loaded data sets.
+       */
+      template <int dim>
+      class AsciiDataLookup
+      {
+        public:
+
+          /**
+           * Initialize all members and the two tables referring to the
+           * actual velocities.
+           */
+          AsciiDataLookup(const Point<dim> &grid_extent_,
+                          const unsigned int npoints_[3],
+                          const unsigned int boundary_dimensions_[2],
+                          const double scale_factor_);
+
+          /**
+           * Outputs the AsciiData module information at model start. Need to be
+           * separated from Constructor because at construction time the
+           * SimulatorAccess is not initialized and only Rank 0 should give
+           * the screen output.
+           */
+          void screen_output(const ConditionalOStream &pcout) const;
+
+          /**
+           * Check whether a file named filename exists.
+           *
+           * @param filename File to check existence
+           */
+          bool fexists(const std::string &filename) const;
+
+          /**
+           * Loads a text velocity file. Throws an exception if the
+           * file does not exist, if the velocity file format is not correct or
+           * if the number of read in points does not match number of assumed
+           * points.
+           */
+          void load_file(const std::string &filename,
+                         const ConditionalOStream &pcout);
```

Comment:
I think it would probably be easier to understand the function if it didn't take the second argument and let the caller generate the output. It's true that that requires some code duplication, but a function that's called `load_file` should in fact just do that. Or you could rename it as `load_file_and_print_status`, but I think that's awkward too :-)